lib/facter/framework/core/fact/internal/internal_fact_manager.rb

```diff
@@ -47,5 +55,40 @@ def resolve(searched_facts)
       resolved_facts.flatten!
       FactAugmenter.augment_resolved_facts(searched_facts, resolved_facts)
     end
+
+    def start_threads(searched_facts)
+      threads = []
+      # only resolve a fact once, even if multiple search facts depend on that fact
+      searched_facts
+        .uniq { |searched_fact| searched_fact.fact_class.name }
+        .each do |searched_fact|
+        threads << Thread.new do
+          resolve_fact(searched_fact)
+        end
+      end
+
+      threads
+    end
+
+    def join_threads(threads, searched_facts)
+      resolved_facts = []
+
+      threads.each do |thread|
+        thread.join
```

Comment:
i think Thread#join is not needed as Thread#value(https://ruby-doc.org/core-2.5.1/Thread.html#method-i-value) is using join.

Reply:
Removed join